src/depcache.erl

```diff
@@ -696,61 +717,62 @@ find_value(_Key, _Data) ->
     end;
 
 %% After finishing a batch in cache_full mode, check if the cache is still full, if so keep deleting entries
-cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr, Now, cache_full, 0) -> 
-    case cleanup_mode(DataTable, MemoryMax) of
-        normal     -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr, Now, normal, 0);
-        cache_full -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr, now_sec(), cache_full, ?CLEANUP_BATCH)
+cleanup(#cleanup_state{} = State, SlotNr, Now, cache_full, 0) ->
+    case cleanup_mode(State) of
+        normal     -> ?MODULE:cleanup(State, SlotNr, Now, normal, 0);
+        cache_full -> ?MODULE:cleanup(State, SlotNr, now_sec(), cache_full, ?CLEANUP_BATCH)
     end;
 
 %% Normal cleanup behaviour - check expire stamp and dependencies
-cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr, Now, normal, Ct) ->
-    Slot =  try 
+cleanup(#cleanup_state{tables = #tables{meta_table = MetaTable}} = State, SlotNr, Now, normal, Ct) ->
+    Slot =  try
                 ets:slot(MetaTable, SlotNr)
             catch
                 _M:_E -> '$end_of_table'
             end,
     case Slot of
-        '$end_of_table' -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, '$end_of_table', Now, normal, 0);
-        [] -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr+1, Now, normal, Ct-1);
+        '$end_of_table' -> ?MODULE:cleanup(State, '$end_of_table', Now, normal, 0);
+        [] -> ?MODULE:cleanup(State, SlotNr + 1, Now, normal, Ct - 1);
         Entries ->
-            lists:foreach(fun(Meta) -> flush_expired(Meta, Now, DepsTable, Pid) end, Entries),
-            ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr+1, Now, normal, Ct-1)
+            lists:foreach(fun(Meta) -> flush_expired(Meta, Now, State) end, Entries),
+            ?MODULE:cleanup(State, SlotNr + 1, Now, normal, Ct - 1)
     end;
 
 %% Full cache cleanup mode - randomly delete every 10th entry
-cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr, Now, cache_full, Ct) ->
-    Slot =  try 
+cleanup(#cleanup_state{pid = Pid, name = Name, callback = Callback, tables = #tables{meta_table = MetaTable}} = State, SlotNr, Now, cache_full, Ct) ->
+    Slot =  try
                 ets:slot(MetaTable, SlotNr)
             catch
                 _M:_E -> '$end_of_table'
             end,
     case Slot of
-        '$end_of_table' -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, '$end_of_table', Now, cache_full, 0);
-        [] -> ?MODULE:cleanup(Pid, MetaTable, DepsTable, DataTable, MemoryMax, SlotNr+1, Now, cache_full, Ct-1);
+        '$end_of_table' -> ?MODULE:cleanup(State, '$end_of_table', Now, cache_full, 0);
+        [] -> ?MODULE:cleanup(State, SlotNr + 1, Now, cache_full, Ct - 1);
         Entries ->
+            callback(eviction, Name, Callback),
```

Comment:
Is this the proper place to call the callback? Or should it go [here](https://github.com/zotonic/depcache/pull/4/files#diff-e89b527335efc9f7f45ddb9a2294e3b3R783) instead?